### ISSUE TITLE
com.bugsnag.JavaScriptException Invalid response for blob

### ISSUE THREAD

**Issue by peacechen** [bugsnag/bugsnag-react-native#232]
### Description

On a production build, the app is crashing on resume, generating a Bugsnag report with the breadcrumbs:
```
com.bugsnag.JavaScriptException 0s before | message Invalid response for blob:
com.bugsnag.JavaScriptException 0s before | message Invalid response for blob:
```

I've uploaded sourcemaps for this build but it's not resolving.  This app is using CodePush and I've previously been able to upload CP sourcemaps to Bugsnag.  In this particular case it's a new version that hasn't had a CodePush update deployed.  I wonder if that's confusing the sourcemap resolution.  The top of the stacktrace shows
```
index.android.bundle:382:8651 get
```

### Issue

Is the crash report pointing to Bugsnag itself, or our code?   A previous build of our app was working with Bugsnag.  We recently updated to React Native 0.55.1.

### Environment

Library versions:

```shell
bugsnag-react-native@2.9.3
react-native-code-push@5.3.2
react-native@0.55.1
```

- iOS/Android version(s): Android 7.1.2
- simulator/emulator or physical device?: physical device only
- debug mode or production?: production only

- [x] (Android only) `BugsnagReactNative.start(this)` is present in the
  `onCreate` method of your `MainApplication` class?

### Example code snippet

```js
import { Client, Configuration } from 'bugsnag-react-native';

const config = new Configuration();
config.codeBundleId = appVersion;
config.autoCaptureSessions = true;
config.registerBeforeSendCallback((report, error) => {
	report.metadata = {
		...report.metadata,
		extra: {
			UUID: DeviceInfo.getUniqueID(),
			reactNativeVersion: require('react-native/package.json').version,
			appVersion
		}
	}
});
const bugsnag = new Client(config);
```

**Comment by peacechen** [comment]
Turns out to be a bug in RN 0.54 and 0.55.  Either revert to 0.53, or use my fork which is based on the 0.55.2 stable release with the fix to XMLHttpRequest applied.
https://github.com/peacechen/react-native

**Comment by kattrali** [comment]
Thanks for the report, @peacechen. Do you have a link to the react-native issue? I'm going to close this issue as it isn't actionable for the Bugsnag team, but would like to keep tracking of upstream issues which might be relevant to support.

**Comment by peacechen** [comment]
This issue affects both iOS and Android.  There is a proposed patch which hasn't been merged.
https://github.com/facebook/react-native/issues/18223

@kattrali  Would you know why our sourcemaps aren't resolving?  Our app uses CodePush, and in previous versions with CodePush deployments it has resolved.  In this specific case it was a fresh version with no CodePush updates deployed.  I uploaded sourcemaps with `--code-bundle-id` and `--add-wildcard-prefix` to try to match the CodePush "version" that I configured in the app as e.g. 3.1.2-0 to indicate no CodePush update.  I also tried it without those flags to indicate a plain RN bundle without CodePush.

**Comment by kattrali** [comment]
Thanks, @peacechen. 

> Would you know why our sourcemaps aren't resolving?
Not offhand, but I can look into it. Can you send an email to support@bugsnag.com with this information so we can take a look at your projects/source maps and see what's going wrong in the resolution process?

**Comment by peacechen** [comment]
4 of the 5 crash reports could not resolve the sourcemaps, but the final one did.  Would the order of crash reports and sourcemap uploads make a difference?  Here's what I think happened:

Crash report 1
Crash report 2
...
Uploaded sourcemaps (both CodePush and bare bundles)
Crash report 5

**Comment by yaronlevi** [comment]
Any advances on the issues? 

We are on RN@0.49.0 and trying to figure out if it is a good move to upgrade to 0.54/0.55.

We rely heavily on Bugsnag source mapping in Android, and we can not allow breaking this feature.

**Comment by peacechen** [comment]
I believe the empty blob response errors have been fixed in PRs that will be in RN 0.56.  That's supposed to be released next week or so.

**Comment by yaronlevi** [comment]
So currently Bugsnag won't work with 0.54/0.55?
